src/main/java/org/springframework/retry/policy/BinaryExceptionClassifierRetryPolicy.java

```diff
@@ -28,16 +29,15 @@
  *
  * @author Aleksandr Shamukov
  */
-@SuppressWarnings("serial")
 public class BinaryExceptionClassifierRetryPolicy implements RetryPolicy {
 
-	private final BinaryExceptionClassifier exceptionClassifier;
+	private final Classifier<Throwable, Boolean> exceptionClassifier;
 
-	public BinaryExceptionClassifierRetryPolicy(BinaryExceptionClassifier exceptionClassifier) {
+	public BinaryExceptionClassifierRetryPolicy(Classifier<Throwable, Boolean> exceptionClassifier) {
```

Comment:
While I agree with you, this is a breaking change for the current `2.0.x` version.
There is already the `ExceptionClassifierRetryPolicy` which supports any arbitrary:
```
	/**
	 * Setter for an exception classifier. The classifier is responsible for translating
	 * exceptions to concrete retry policies. Either this property or the policy map
	 * should be used, but not both.
	 * @param exceptionClassifier ExceptionClassifier to use
	 */
	public void setExceptionClassifier(Classifier<Throwable, RetryPolicy> exceptionClassifier) {
```

So, how about to use that one instead in the `RetryTemplateBuilder` when we provide such a `Predicate` you are talking about?

Reply:
I do agree `ExceptionClassifierRetryPolicy` is mighty option, but at the cost of complexity. Simple code like this:
```java
        return RetryTemplate.builder()
            .maxAttempts(3)
            .fixedBackoff(200)
            .retryOn(throwable -> throwable instanceof SQLException sqlException
                && sqlException.getErrorCode() == 256)
            .build();
```
becomes 20+ lines of code making it hard to maintain and error prone :-(

From my point of view, the constructor you address is backward compatible, because you can still use `BinaryExceptionClassifier` instance as an argument. It implements `Classifier<Throwable, Boolean>`

Reply:
But you have modified getter as well, which is not binary compatible.
Plus I didn't mean to reject your suggestion for the `RetryTemplateBuilder` API.
I just suggested to use that `ExceptionClassifierRetryPolicy` in the `build()` when simple `Predicate` is provided.
This way we definitely won't break any API and have a nice addition to the target functionality.

Reply:
You're right about the getter. About ExceptionClassifierRetryPolicy it is my fault. I didn't get your idea. I'll try to update the PR following your advice.

Reply:
After few experiments with ExceptionClassifierRetryPolicy I decided to introduce new PredicateRetryPolicy to produce more simple code. I'll be happy to have your opinion?